### ISSUE TITLE
Expecting "int", got "string" on property Timeout on Lambda Function

### ISSUE THREAD

**Issue by yyolk** [awslabs/aws-sam-local#50]
I'm receiving an error when providing a `Timeout` via a `!Ref`

The resource I'm receiving the error on:

```yaml
 CustomAuthorizerLambdaFunction:
   Type: AWS::Serverless::Function
   Properties:
     CodeUri: ./code.jar
     Handler: someClass
     Runtime: java8
     Timeout: !Ref FunctionTimeout
     VpcConfig:
       SecurityGroupIds:
         - !Ref LambdaVPCSecurityGroup
       SubnetIds: !Ref Subnets
     MemorySize: 512
     Environment:
       Variables:
         REDIS_HOST: !Ref RedisHost
         REDIS_PORT: "6379"
         DEFAULT_DB_TIMEOUT: !Ref DBIdleTimeout
     Policies:
       - AWSLambdaVPCAccessExecutionRole
       - AWSXrayWriteOnlyAccess
     Tags:
       Department: engineering
       Application: customer-api
       Service: authorizer-processing
```

the error I receive: 
```
ERROR: Resource "CustomAuthorizerLambdaFunction", property "Timeout": Expecting "int", got "string" (line: 278; col: 6)
```

**Comment by PaulMaddox** [comment]
Hi @yyolk,

Currently support for [CloudFormation Intrinsic Functions](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) is less than ideal in SAM Local (specifically the [goformation](https://github.com/awslabs/goformation) we're using to parse/validate SAM templates).

This is something we're working hard on improving (see the [0.1.0 branch](https://github.com/awslabs/goformation/tree/0.1.0/) of the goformation library that's currently in-progress, and specifically [this proposal](https://github.com/awslabs/goformation/issues/3)).

In the meantime, in order to use SAM Local with this template, you'll have to remove the `!Ref` intrinsic functions, and just replace them with integer values.

**Comment by sanathkr** [comment]
Fixed in [v0.2.0](https://github.com/awslabs/aws-sam-local/releases/tag/v0.2.0)
